examples/step-90/step-90.cc

```diff
@@ -0,0 +1,1271 @@
+        {
+          error_L2_sqr += copy_data.cell_L2_error_sqr;
+          error_H1_sqr += copy_data.cell_H1_error_sqr;
+          error_stab_sqr += copy_data.cell_stab_sqr;
+        }
+    };
+
+    ScratchData<dim> scratch_data(mapping,
+                                  fe_collection,
+                                  mesh_classifier,
+                                  level_set_dof_handler,
+                                  level_set,
+                                  surface_update_flags,
+                                  cell_quadrature,
+                                  quadrature_1D);
+
+    CopyData<dim> copy_data;
+
+    MeshWorker::mesh_loop(dof_handler.begin_active(),
+                          dof_handler.end(),
+                          cell_worker,
+                          copier,
+                          scratch_data,
+                          copy_data,
+                          MeshWorker::assemble_own_cells);
+    timer.stop();
+    pcout << "took (" << timer.wall_time() << "s)" << std::endl;
+  }
+
+  template <int dim>
+  void LaplaceBeltramiSolver<dim>::display_results()
+  {
+    const double error_L2 =
+      std::sqrt(Utilities::MPI::sum(error_L2_sqr, mpi_communicator));
+    const double error_semiH1 =
+      std::sqrt(Utilities::MPI::sum(error_H1_sqr, mpi_communicator));
+    const double error_stab =
+      std::sqrt(Utilities::MPI::sum(error_stab_sqr, mpi_communicator));
+
+    const int dof_handler_size           = dof_handler.n_dofs();
+    const int level_set_dof_handler_size = level_set_dof_handler.n_dofs();
+
+    const double iterations =
+      Utilities::MPI::max(number_of_iterations, mpi_communicator);
+
+    convergence_table.add_value("LevelSet dofs", level_set_dof_handler_size);
+    convergence_table.evaluate_convergence_rates(
+      "LevelSet dofs", ConvergenceTable::reduction_rate_log2);
+
+    convergence_table.add_value("Active dofs", dof_handler_size);
+    convergence_table.evaluate_convergence_rates(
+      "Active dofs", ConvergenceTable::reduction_rate_log2);
+
+    convergence_table.add_value("Iterations", iterations);
+
+    convergence_table.add_value("L2 Error", error_L2);
+    convergence_table.evaluate_convergence_rates(
+      "L2 Error", ConvergenceTable::reduction_rate_log2);
+    convergence_table.set_scientific("L2 Error", true);
+
+    convergence_table.add_value("H1 error", error_semiH1);
+    convergence_table.evaluate_convergence_rates(
+      "H1 error", ConvergenceTable::reduction_rate_log2);
+    convergence_table.set_scientific("H1 error", true);
+
+    convergence_table.add_value("Stab norm", error_stab);
+    convergence_table.evaluate_convergence_rates(
+      "Stab norm", ConvergenceTable::reduction_rate_log2);
+    convergence_table.set_scientific("Stab norm", true);
+
+    pcout << std::endl;
+    if (Utilities::MPI::this_mpi_process(mpi_communicator) == 0)
+      convergence_table.write_text(pcout.get_stream());
```

Comment:
If you move the different parts of this function into other places as suggested, you're really just left with these two lines. You could then just move this into `run()` where you currently call the `display_results()` function.